ktor-client/ktor-client-curl/src/io/ktor/client/engine/curl/CurlCallbacks.kt

```diff
@@ -0,0 +1,72 @@
+package io.ktor.client.engine.curl
+
+import kotlinx.cinterop.*
+import libcurl.*
+import libcurl.CURLMsg.*
+import platform.posix.size_t
+import platform.posix.memcpy
+
+private fun CPointer<ByteVar>.toKString(length: Int) = this.readBytes(length).stringFromUtf8()
```

Comment:
obsolete this

Reply:
Why we hardcode utf8 here? Should we provide charset argument instead? 
We could use some decoder utility functions in kotlinx.io